src/features.jl

```diff
@@ -0,0 +1,69 @@
+_flattenimage(image::Matrix) = vec(image)
+
+function _flattenimage{T}(images::Array{T,3})
+    ncols, nrows, nimages = size(images)
+    reshape(images, (nrows * ncols, nimages))
+end
+
+
+feature_doc(train_or_test) =
+"""
+    $(train_or_test)features([indices])
+
+Returns the MNIST **$(train_or_test)set** features denoted by `indices`.
+
+- if `indices` is an `Integer`, the single image is returned as
+flat `Vector{Float64}` feature vector.
+
+- if `indices` is a `AbstractVector`, the images are returned as
+a design matrix `Matrix{Float64}` in feature major layout, which
```

Comment:
Pedantic nit: should we call this a matrix rather than a design matrix? The Wikipedia article suggests that a design matrix is stored with features as columns, although it's not clear to me where they get the justification for that claim.